File: 10_week_2/second_largest.js
// given an array, return the second-largest element

function second_largest(arr) {
	if (arr.length < 2) {
		return null;
	}
	var max = arr[0], second_max;
	for (var i = 1, j = arr.length; i < j; i++) {
		if (max < arr[i]) {
			if (second_max == undefined) {
				second_max = max;
			}
			else if (second_max < max) {
				second_max = max;
			}
			max = arr[i];
		}
		else if (second_max == undefined) {
			second_max = arr[i];
		}
		else if (second_max < arr[i]) {
			second_max = arr[i];
		}
		// console.log("current max: ", max);
		// console.log("current second max: ", second_max);
	}
	return second_max;
}

var arr = [23, 32, -2 , 6 ,2 ,7, 10, 3, 10, 22, 3, -1, 19, 24, 33];
var result = second_largest(arr);
console.log(result);

File: 10_week_2/second_largest.py
# given an array, return the second-largest element

def second_largest(arr):
	if len(arr) - 2 < 0:
		return None
	max = arr[0]
	second_max = "undefined"
	for idx in range(1, len(arr)):
		if max < arr[idx]:
			if second_max == "undefined":
				second_max = max
			elif second_max < max:
				second_max = max
			max = arr[idx]
		elif second_max == "undefined":
			second_max = arr[idx]
		elif second_max < arr[idx]:
			second_max = arr[idx]
	return second_max

arr = [23, 32, -2 , 6 ,2 ,7, 10, 3, 10, 22, 3, -1, 19, 24, 33]
result = second_largest(arr)
print(result)
